scripts/imgtool.py

```diff
@@ -116,8 +116,10 @@ def convert(self, value, param, ctx):
 @click.argument('infile')
 @click.option('-M', '--max-sectors', type=int,
               help='When padding allow for this amount of sectors (defaults to 128)')
-@click.option('--pad', type=BasedIntParamType(),
-              help='Pad image to this many bytes, adding trailer magic')
+@click.option('--pad', default=False, is_flag=True,
+              help='Pad image to --slot-size bytes, adding trailer magic')
+@click.option('-S', '--slot-size', type=BasedIntParamType(), required=True,
```

Comment:
This is required. I assume that intentionally (so need a lite more knowledge for a newcomer). I think it should be still worth to make this optional.

Reply:
Yes, it is intentional, making it optional just means nobody will used it.

Reply:
OK - reasonable, good job